emperor/support_files/js/sceneplotview3d.js

```diff
@@ -100,6 +102,90 @@ define([
     this.dimensionRanges = {'max': [], 'min': []};
     this.drawAxesWithColor(0xFFFFFF);
     this.drawAxesLabelsWithColor(0xFFFFFF);
+
+    var raycaster = new THREE.Raycaster();
+    var mouse = new THREE.Vector2();
+
+    //initialize subscribers for event callbacks
+    this.EVENTS = ['click', 'dblclick'];
+    var subscribers = {};
+    for (var i = 0; i < this.EVENTS.length; i++) {
+      subscribers[this.EVENTS[i]] = [];
+    }
+
+    //create helper function for callbacks
+    var eventCallback = function(eventType, event) {
+      event.preventDefault();
+      //dont do anything if no subscribers
+      if (subscribers[eventType].length === 0) {
+        return;
+      }
+
+      var element = scope.renderer.domElement;
+      mouse.x = ((event.clientX - element.offsetLeft) / element.width) * 2 - 1;
+      mouse.y = -((event.clientY - element.offsetTop) / element.height) * 2 + 1;
+
+      raycaster.setFromCamera(mouse, scope.camera);
```

Comment:
I lied. Also very relevant. Uses the marquee selection from the last comment, and also shows how to combine caching

https://github.com/cubicleDowns/webgl-code-samples/tree/master/canvas-interaction